### DISCUSSION TITLE
Increasing protocolTimeout?

### DISCUSSION THREAD

**Discussion by mrcrmn** [spatie/browsershot#760] [Q&A]
Hello everyone,

im trying to generate a very large PDF file and I'm running into `protocolTimeout` Errors.

I can increase that timeout by editing that property in the `bin/browser.js` file directly but I was wondering if I can accomplish the the same, without editing vendor files 😄 

Thanks alot!

**Comment by nbyloff** [comment]
I just came here to ask the same question. Did you ever find a solution?

**Reply by mrcrmn** [reply]
Hi @nbyloff 

Unfortunately not, I now generate multiple, smaller PDF chunks and then merge them together via Ghostscript.

**Comment by zarulizham** [comment]
Can I create PR for this? Let say we set 30s for default `protocolTimeout` value. 

Inside bin/browser.cjs:
```js
browser = await puppet.launch({
    ...
    ...
    ...
    protocolTimeout: request.options.protocolTimeout,
});
```

src/Browsershot.php:
```php
protected int $protocolTimeout = 30;

public function protocolTimeout(int $protocolTimeout): static
{
    $this->protocolTimeout = $protocolTimeout;

    return $this->setOption('protocolTimeout', $protocolTimeout * 1000);
}
```

**Reply by zarulizham** [reply]
https://github.com/spatie/browsershot/pull/865

PR has been merged.
